### PR TITLE
Rubocop jekyll.rb

### PR DESCRIPTION
**Pull request by floydpraveen**
fixes one file from #4964 
#### work in progress
yet to fix below offence reported by rubocop
```
lib/jekyll.rb:119:9: C: Do not prefix writer method names with set_.
    def set_timezone(timezone)
        ^^^^^^^^^^^^
```

### CODE REVIEW

**Comment by envygeeks** [comment]
hey @floydpraveen do you mind adding `# rubocop:disable Style/AccessorMethodName` above `set_timezone` and then below it and above the next method name do `# rubocop:enable Style/AccessorMethodName`

This is :shipit: after that minor change!
Thanks!

**Comment by floydpraveen** [comment]
@envygeeks done, thanks for reviewing it

**Comment by floydpraveen** [comment]
@envygeeks let me reword the commit messages

File: lib/jekyll.rb [hunk 1]
```diff
@@ -115,9 +116,11 @@ def configuration(override = Hash.new)
     # timezone - the IANA Time Zone
     #
     # Returns nothing
+    # rubocop:disable Style/AccessorMethodName
     def set_timezone(timezone)
-      ENV['TZ'] = timezone
+      ENV["TZ"] = timezone
     end
+    # rubocop:enable Style/AccessorMethodName
```
Comment (parkr) [inline lib/jekyll.rb]: I don't think this is necessary 😄  It's just applies to the method when it's right above the method like that, I'm pretty sure.

File: lib/jekyll.rb [hunk 2]
```diff
@@ -154,29 +157,29 @@ def sites
     def sanitized_path(base_directory, questionable_path)
       return base_directory if base_directory.eql?(questionable_path)
 
-      questionable_path.insert(0, '/') if questionable_path.start_with?('~')
+      questionable_path.insert(0, "/") if questionable_path.start_with?("~")
       clean_path = File.expand_path(questionable_path, "/")
-      clean_path.sub!(/\A\w\:\//, '/')
+      clean_path.sub!(%r!\A\w\:\/!, "/")
```
Comment (parkr) [inline lib/jekyll.rb]: Correct me if I'm wrong, @envygeeks, but I think this is `%r!\A\w\:/!`.

File: lib/jekyll.rb [hunk 3]
```diff
@@ -154,29 +157,29 @@ def sites
...
 
-      if clean_path.start_with?(base_directory.sub(/\A\w\:\//, '/'))
+      if clean_path.start_with?(base_directory.sub(%r!\A\w\:\/!, "/"))
```
Comment (parkr) [inline lib/jekyll.rb]: Same regexp here as above, I guess? Maybe we can pull that out into a constant.

**Comment by parkr** [comment]
Looking pretty good! Just needs a rebase and I left a couple comments above. ❤️

File: lib/jekyll.rb (hunk 1 above)
Comment (envygeeks) [inline lib/jekyll.rb]: @parkr it continues for the entire file, though you are right that it isn't necessary.

File: lib/jekyll.rb (hunk 2 above)
Comment (envygeeks) [inline lib/jekyll.rb]: `%r!\A\w:/!`

File: lib/jekyll.rb (hunk 1 above)
Comment (floydpraveen) [inline lib/jekyll.rb]: @parkr @envygeeks since it continues for the entire file, let it be there ? so new method added would be checked by rubocop

**Comment by floydpraveen** [comment]
@parkr @envygeeks made the changes, thanks for comments

File: lib/jekyll.rb [hunk 4]
```diff
@@ -154,29 +157,31 @@ def sites
...
       return base_directory if base_directory.eql?(questionable_path)
 
-      questionable_path.insert(0, '/') if questionable_path.start_with?('~')
+      questionable_path.insert(0, "/") if questionable_path.start_with?("~")
       clean_path = File.expand_path(questionable_path, "/")
-      clean_path.sub!(/\A\w\:\//, '/')
+      clean_path.sub!(%r!\A\w:/!, "/")
 
-      if clean_path.start_with?(base_directory.sub(/\A\w\:\//, '/'))
+
```
Comment (envygeeks) [inline lib/jekyll.rb]: This is causing your formatting specs to fail: you have two new lines instead of one.

File: lib/jekyll.rb (hunk 1 above)
Comment (envygeeks) [inline lib/jekyll.rb]: @floydpraveen yessum

File: lib/jekyll.rb (hunk 4 above)
Comment (floydpraveen) [inline lib/jekyll.rb]: removed it

File: lib/jekyll.rb (hunk 1 above)
Comment (pathawks) [inline lib/jekyll.rb]: @floydpraveen :+1:

**Comment by envygeeks** [comment]
@jekyllbot: merge +dev

**Pull request merged**